kuma/wiki/utils.py

```diff
@@ -59,3 +69,80 @@ def tidy_content(content):
         return tidied.decode('utf-8'), errors
     else:
         return content
+
+
+def analytics_upageviews(revision_ids, start_date, end_date=None):
+    """Given a sequence of document revision IDs, returns a dict matching
+    those with the number of users Google Analytics thinks has visited
+    each revision since start_date.
+
+    """
+
+    scopes = ['https://www.googleapis.com/auth/analytics.readonly']
+
+    try:
+        ga_cred_dict = json.loads(config.GOOGLE_ANALYTICS_CREDENTIALS)
+    except (ValueError, TypeError):
+        raise ImproperlyConfigured(
+            "GOOGLE_ANALYTICS_CREDENTIALS Constance setting is badly formed.")
+    if not ga_cred_dict:
+        raise ImproperlyConfigured(
+            "An empty GOOGLE_ANALYTICS_CREDENTIALS Constance setting is not permitted.")
+
+    credentials = ServiceAccountCredentials.from_json_keyfile_dict(ga_cred_dict,
+                                                                   scopes=scopes)
+    http_auth = credentials.authorize(Http())
+    service = build('analyticsreporting', 'v4', http=http_auth)
+
+    end_date = datetime.date.today().isoformat()
+    if hasattr(start_date, 'date'):
+        start_date = start_date.date()
```

Comment:
👍 on auto-fixing ``datetime`` to ``date``. The same should be used on ``end_date``.